lib/FlashMem/FlashMem.cpp

```diff
@@ -0,0 +1,191 @@
+
+/******************************************************************************
+ * Prototypes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Local Variables
+ *****************************************************************************/
+
+/******************************************************************************
+ * Public Methods
+ *****************************************************************************/
+
+bool Flash::begin()
+{
+    bool isSuccess = true;
+    EEPROM.begin(EEPROM_SIZE);
+    return isSuccess;
+}
+
+bool Flash::importCredentials(String &ssid, String &password)
+{
+    bool isSuccess = false;
+
+    if (areCredentialsStored())
+    {
+        char temp = 0x00;
+
+        for (uint8_t memoryPosition = 0; memoryPosition <= NVM_SSID_MAX_LENGTH; memoryPosition++)
+        {
+            temp = EEPROM.read(NVM_SSID_ADDRESS + memoryPosition);
+
+            if (0x00 == temp)
+            {
+                break;
+            }
+            else
+            {
+                ssid += temp;
+            }
+        }
+
+        if (0x00 != EEPROM.read(NVM_PASSWORD_ADDRESS))
+        {
+            for (uint8_t memoryPosition = 0; memoryPosition <= NVM_PASSWORD_MAX_LENGTH; memoryPosition++)
+            {
+                temp = EEPROM.read(NVM_PASSWORD_ADDRESS + memoryPosition);
+
+                if (0x00 == temp)
+                {
+                    break;
+                }
+                else
+                {
+                    password += temp;
+                }
+            }
+        }
+
+        Serial.println("\n SSID: " + ssid + "\n Password: " + password);
+
+        isSuccess = true;
+    }
+
+    return isSuccess;
+}
+
+bool Flash::saveCredentials(const String &ssid, const String &password)
+{
+    bool isSuccess = false;
+
+    Flash::clearEEPROM();
+
+    if ((!ssid.isEmpty()) && (ssid.length() <= CREDENTIALS_MAX_LENGTH))
+    {
+        /* Save SSID */
+        for (uint8_t memoryPosition = 0; memoryPosition <= ssid.length(); memoryPosition++)
+        {
+            EEPROM.write(NVM_SSID_ADDRESS + memoryPosition, ssid[memoryPosition]);
+            Serial.print(ssid[memoryPosition]);
+        }
+        Serial.println();
+
+        if ((!password.isEmpty()) && (password.length() <= CREDENTIALS_MAX_LENGTH))
+        {
+            /* Save Password */
+            for (uint8_t memoryPosition = 0; memoryPosition <= password.length(); memoryPosition++)
+            {
+                EEPROM.write(NVM_PASSWORD_ADDRESS + memoryPosition, password[memoryPosition]);
+                Serial.print(password[memoryPosition]);
+            }
+            Serial.println();
+        }
+
+        /* Write 0x55 and 0xA5 to Metadata Header are Credentials are Stored) */
+        EEPROM.write(NVM_METADATA_ADDRESS, 0x55);
+        EEPROM.write(NVM_METADATA_ADDRESS + 1, 0xA5);
```

Comment:
Magic magic .... If I change one of them, at which places I have to change it too? Keep that in mind.